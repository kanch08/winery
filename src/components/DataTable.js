import React, {Component} from 'react';
import TableRow from './TableRow';
import '../components/stylesheet/tableItem.css';

class DataTable extends Component {

    render() {
        const {items} = this.props;
        const addedItems = items.addedItems || [];

        //  console.log('items in data table',items);


        return (

            <div className="tableData">


                <table className="final-table">
                    <thead>
                    <tr>
                        <th> Id</th>
                        <th>Name</th>
                        <th>Price</th>
                        <th>Color</th>
                        <th>Size</th>
                        <th>Quantity</th>



                    </tr>
                    </thead>
                    <tbody>

                    {
                        addedItems.map(item => {
                                return (
                                    <TableRow key={item.id} item={item}/>
                                )

                            }
                        )
                    }
                    </tbody>


                </table>
            </div>

        );
    }
}

export default DataTable;