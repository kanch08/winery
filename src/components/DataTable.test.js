import React from 'react';
import { mount, shallow } from 'enzyme';
import DataTable from "./DataTable";
import toJson from "enzyme-to-json";

describe('testing Data table component',  () => {
    let spy;
    const addedItems = {
        name: 'kanchan',
        work: 'SE',
    }
    it('test props of DataTable', () => {

        const wrapper= mount(<DataTable  items={addedItems}/>);
        expect(wrapper.props().items).toEqual(addedItems);
    })
    it('test the componentDid Mount', () => {
        const addedItems = [
        { name: 'kanchan'},
        {work: 'SE'},
    ]
        spy = jest.spyOn(DataTable.prototype, 'render');
        const wrapper = mount(<DataTable {...addedItems}/>);
        wrapper.instance();
        expect(spy).toHaveBeenCalled();
    })
    it('renders correctly', () => {
        const wrapper = shallow(<DataTable items={addedItems}/>)
        expect(toJson(wrapper)).toMatchSnapshot();
    })

})