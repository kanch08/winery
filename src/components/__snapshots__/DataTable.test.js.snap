// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Data table component renders correctly 1`] = `
<div
  className="tableData"
>
  <table
    className="final-table"
  >
    <thead>
      <tr>
        <th>
           Id
        </th>
        <th>
          Name
        </th>
        <th>
          Price
        </th>
        <th>
          Color
        </th>
        <th>
          Size
        </th>
        <th>
          Quantity
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
